### PR TITLE
Fix an issue with using postgres and add postgres config

### PR DESCRIPTION
**Pull request by jahmai-ca**
This enables and allows easy switching between Cosmos and Postgres and restores a couple of x-partition query helpers in lieu of further fixes.

### CODE REVIEW

File: dendrite-config-cosmosdb.yaml [hunk 1]
```diff
@@ -389,6 +389,6 @@ tracing:
 # stdout by Dendrite.
 logging:
 - type: file
-  level: info
+  level: debug
```
Comment (alexfca) [inline dendrite-config-cosmosdb.yaml:392]: Do we wanna keep this?

**Review by alexfca — APPROVED** [review]

File: dendrite-config-cosmosdb.yaml (hunk 1 above)
Comment (jahmai-ca) [reply dendrite-config-cosmosdb.yaml:392]: Yeah not sure why not, this is only for local dev right? can't think of a reason not to have to set to debug for local dev? Happy to change it back tho

Comment (alexfca) [reply dendrite-config-cosmosdb.yaml:392]: it spits out heaps more data...
i think this is the template @jaredjn uses when provisioning, but not 100% sure.
When we are running (ie without issues) then Info would be preferrable.
If we have probs, then a dev can turn on Debug to see whats going on, so would prefer Info at a min by default

Comment (jahmai-ca) [reply dendrite-config-cosmosdb.yaml:392]: Oh its a template? Ok I'll remove

Comment (jahmai-ca) [reply dendrite-config-cosmosdb.yaml]: Done

**Pull request merged**